persistence/remote/src/main/java/org/infinispan/persistence/remote/upgrade/HotRodTargetMigrator.java

```diff
@@ -80,22 +90,57 @@ public void run() {
                      } catch (Exception e) {
                         log.keyMigrationFailed(Util.toStr(key), e);
                      }
-                  }
-               });
+                  });
+
+               }
+               es.shutdown();
+               try {
+                  while (!es.awaitTermination(500, TimeUnit.MILLISECONDS)) ;
+               } catch (InterruptedException e) {
+                  throw new CacheException(e);
+               }
+               return count.longValue();
+            }
+         } else {
+            final AtomicInteger count = new AtomicInteger(0);
+            ExecutorService executorService = Executors.newFixedThreadPool(threads);
 
+            try (CloseableIterator<List<Entry<Object, Object>>> iterator = new GroupedIterator<>(
+                    storeCache.retrieveEntries(null, null, DEFAULT_READ_BATCH_SIZE), DEFAULT_WRITE_BATCH_SIZE)) {
+               while (iterator.hasNext()) {
+                  Map<Object, Object> map = new HashMap<>();
+                  for (Entry<Object, Object> entry : iterator.next()) {
```

Comment:
Unfortunately the iterator loses the metadata (i.e. expiration time), which is instead preserved by the RCS when using rawValues, since it invokes the remoteCache.getWithMetadata() op